### DISCUSSION TITLE
`in` filter for json columns.

### DISCUSSION THREAD

**Discussion by shobhit1997** [prisma/prisma#19465] [Q&A]
I want to do something like below. How can I achieve this on prisma. Currently it is not supported it seems.

```
 await this.prismaService.eventCandidate.findMany({
        where: {
          registration_data: {
            path: ['Gender'],
            in: ['Male', 'Female'],
          },
        },
      })
```

**Comment by ludralph** [comment]
Hi @shobhit1997 👋 

The `in` filter isn't supported for JSON columns. These are a list of the allowable filter:
```js
type JsonNullableFilter {
  equals?: Json | JsonNullValueFilter
  path?: List<String>
  string_contains?: String
  string_starts_with?: String
  string_ends_with?: String
  array_contains?: Json | Null
  array_starts_with?: Json | Null
  array_ends_with?: Json | Null
  lt?: Json
  lte?: Json
  gt?: Json
  gte?: Json
  not?: Json | JsonNullValueFilter
}
```

**Reply by shobhit1997** [reply]
Yes @ludralph , I know that, but I have a requirement like that. Following query is actually supported by Postgres. `Select * from event_candidates where "registration_data"->>'Gender' IN ('Male', 'Female');` . How can I write it in Prisma, when in is not available ?

**Reply by ludralph** [reply]
Hi @shobhit1997 You can write it using an exhaustive OR condition as suggested in this open [issue](https://github.com/prisma/prisma/issues/10397)
```js
await this.prismaService.eventCandidate.findMany({
    where:{
      OR: [
        { registration_data: { path: ['Gender'], equals: 'Male'} },
        { registration_data: { path: ['Gender'], equals: 'Female'} },
    ]
    }
  })
```
or you can choose to use [raw query](https://www.prisma.io/docs/concepts/components/prisma-client/raw-database-access)

**Reply by ludralph** [reply]
Hi @shobhit1997 

Do you have any more questions around this issue? If no, it would be really helpful if you could quickly **mark this Discussion as _answered_** to indicate that your issue has been resolved. 

In case you do have any outstanding questions, feel free to let us know and we'd be happy to help!

**Reply by nurul3101** [reply]
We are closing this Discussion due to inactivity. Please open a new Discussion if you have a similar problem 🙏
